packages/ingest/granule.js

```diff
@@ -96,6 +97,25 @@ class Discover {
     return this.collection.files.find((fileTypeConfig) => file.name.match(fileTypeConfig.regex));
   }
 
+  /**
+   * Determine if a granule does not yet exist in Cumulus
+   *
+   * @param {Object} granule - the granule that's being looked for
+   * @param {string} granule.name - the name of the granule in Cumulus
+   * @returns {Promise.<(null|Object)>} - a Promise that resolves to null
+   *   when the object does already exist in Cumulus, or the passed-in granule
+   *   object if it does not already exist.
+   */
+  async granuleIsNew(granule) {
+    const event = {
+      pathParameters: { granuleName: granule.name }
+    };
+
+    const response = await getGranule(event);
+    if (response.statusCode === 404) return granule;
```

Comment:
Wouldn't we return null if the status code is a 404?

Reply:
Great question. `getGranule` will throw a 404 if it doesn't find the granule in dynamodb. If the granule does exist, then we return `null`, otherwise we'll return the granule object passed into `granuleIsNew`. The functionality isn't really intuitive and I think improvements could be made.